### PR TITLE
Make the SDK a go module

### PR DESCRIPTION
**Pull request by shigmas**
modules are great!!  Well, at least, pretty good.

Created the go.mod, let the go.sum get generated, and the mod tidy'ed it.

### CODE REVIEW

**Comment by johnrichardrinehart** [comment]
Thanks for taking care of this.

**Review by johnrichardrinehart — APPROVED** [review]
I approve.

**Comment by ethankan** [comment]
Todo:

- [x] Remove glide.yaml
- [x] Update instructions in README

**Review by takeshinoda — APPROVED** [review]
LGTM

**Comment by johnrichardrinehart** [comment]
I have pushed a commit to show that I read it, including a few minor changes. It looks good to me.

Thank you.

File: README.md [hunk 1]
```diff
@@ -8,23 +8,24 @@ being released as a public Go module that provides a Go API for devices to inter
...
+If you are not yet using Go modules and are still in GOPATH mode, you may use
 'go get' to fetch the SDK:
 
     $ go get github.com/moderepo/device-sdk-go
 
 ## Using the SDK
 
-The default package name is `mode`. A trivial example:
-
+The package name is `mode` does not match the import path. For example:
```
Comment (shigmas) [inline README.md]: This doesn't sound quite right?

**Pull request merged**